### ISSUE TITLE
send_message is not firing

### ISSUE THREAD

**Issue by markb-trustifi** [KartikTalwar/gmail.js#439]
send_message event is not firing:
v: 0.6.13

```JS
const GmailFactory = require("gmail-js");
const gmail = new GmailFactory.Gmail(window.$);
window.gmail = gmail;
window.addEventListener("load", function() {
   gmail.observe.on("load", () => {
      gmail.observe.on("http_event", function(params) {
            console.log("data:", params);   //working!
        });
       gmail.observe.on("compose", function(params) {
            console.log("data:", params);   //working!
        });
        gmail.observe.on("send_message", function(url, body, data, response, xhr) {
            console.log("on_send_message", "url:", url, 'body', body, 'email_data', data, 'response', response, 'xhr', xhr);   //doesn't work!
        });
        gmail.observe.after("send_message", function(url, body, data, response, xhr) {
            console.log("after_send_message", "url:", url, 'body', body, 'email_data', data, 'response', response, 'xhr', xhr);   //doesn't work!
        });
        gmail.observe.before('send_message', function(url, body, data, xhr){
            console.log("before_send_message", "url:", url, 'body', body, 'email_data', data, 'response', response, 'xhr', xhr);   //doesn't work!
        });
   });
});
```

**Comment by josteink** [comment]
I can reproduce this issue, but not sure why it fails.

Do you know if this is something which used to work for you, which is now broken in a recent release?

From what I can tell, there's no XHR request sent by gmail when a user click "Send" in the compos-window. If this is a change on Gmail's end, we will have a hard time detecting these events in the future.

**Comment by markb-trustifi** [comment]
It is first gmail-js version I'm working with.

**Comment by josteink** [comment]
Right. I've tested now with my production extension and the event fires there, just like it should.

But with your example, Gmail didn't even *send* the email. Based on that, I concluded that your usage of the events had to be wrong, effectively breaking Gmail in the process.

So I tried something simpler:

````javascript
gmail.observe.on("send_message", console.log);
gmail.observe.after("send_message", console.log);
gmail.observe.before('send_message', console.log);
````

And voila, it works:

![image](https://user-images.githubusercontent.com/411338/36085765-841ee678-0fc8-11e8-9877-a959c2ba280f.png)


Your example includes a `response` parameter, and that's nowhere to be found in the definition. If you use TypeScript you will avoid issues like this and the TypeScript-compiler will tell you before you're even running the code :)

The correct signature is

````typescript
private onMessageSent(url: string, body: string, data: any, xhr: XMLHttpRequest): void {
   // your code here
}
````

Basically, omit the `response` parameter, and you should be fine.

**Comment by terance-edmonds** [comment]
Hey there,
It does work fine, my mistake was I  wasn't building the project before testing it once it is updated.
